packages/wonder-blocks-form/src/components/text-area.tsx

```diff
@@ -0,0 +1,416 @@
+     * behaviour. For more details, see the [CSS resize property values MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/resize#values)
+     */
+    resizeType?: "horizontal" | "vertical" | "both" | "none";
+    /**
+     * Change the default focus ring color to fit a dark background.
+     */
+    light?: boolean;
+};
+
+const defaultErrorMessage = "This field is required.";
+
+const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
+    function TextArea(
+        props: TextAreaProps,
+        ref: React.ForwardedRef<HTMLTextAreaElement>,
+    ) {
+        const {
+            onChange,
+            value,
+            placeholder,
+            disabled,
+            id,
+            testId,
+            style,
+            readOnly,
+            autoComplete,
+            name,
+            className,
+            autoFocus,
+            rows,
+            spellCheck,
+            wrap,
+            minLength,
+            maxLength,
+            onClick,
+            onKeyDown,
+            onKeyUp,
+            onFocus,
+            onBlur,
+            validate,
+            onValidate,
+            required,
+            resizeType,
+            light,
+            // Should only include aria related props
+            ...otherProps
+        } = props;
+
+        const [error, setError] = React.useState<string | null>(null);
+
+        const ids = useUniqueIdWithMock("text-area");
+        const uniqueId = id ?? ids.get("id");
+
+        const handleChange = (
+            event: React.ChangeEvent<HTMLTextAreaElement>,
+        ) => {
+            const newValue = event.target.value;
+            onChange(newValue);
+            handleValidation(newValue);
+        };
+
+        const handleValidation = (newValue: string) => {
+            if (validate) {
+                const error = validate(newValue) || null;
+                setError(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+            } else if (required) {
+                const requiredString =
+                    typeof required === "string"
+                        ? required
+                        : defaultErrorMessage;
+                const error = newValue ? null : requiredString;
+                setError(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+            }
+        };
+
+        useOnMountEffect(() => {
+            // Only validate on mount if the value is not empty. This is so that fields
+            // don't render an error when they are initially empty
+            if (value !== "") {
+                handleValidation(value);
+            }
+        });
+
+        // Processing style prop so we can apply aphrodite styles to the textarea
+        // element. The textarea element is used directly so that we can apply inline
+        // styles for width
+        const processedStyle = processStyleList(style);
```

Comment:
Similar to the TextField component, the TextArea component by default has `width: 100%`. When we provide this style to the textarea using Aphrodite, the `!important` property is applied to it. When the built-in browser resize control is used, the width stays at 100% because the Aphrodite style overrides the dynamic width set by the browser.

To get around this, we can use `<textarea>` directly and include an inline style with `width: 100%`. This enables the resize functionality. When we do this, the `style` prop for custom styling can't be applied directly to the `<textarea>` element. To continue to support the `style` prop, we process it and use the resulting `style` with the `css` function from Aphrodite, and the resulting `className` is applied to the `className` prop. The [addStyle utility](https://github.com/Khan/wonder-blocks/blob/4bebd48fb38ae1236394197ea716646b54797aab/packages/wonder-blocks-core/src/util/add-style.tsx#L39-L40) is also using `processStyleList` (that's how I found it 😄 )

For reference, the diff for this change can be found in this commit: https://github.com/Khan/wonder-blocks/pull/2275/commits/d604a49bb8718c9d2123bb6e2f1a416b478ba1e0 .

Let me know if there are any concerns with this or if there is another way to achieve applying non-important styles!

Reply:
**thought:** I'm still not sure about this implementation. I think there's a different way to tackle this:

Moving the `width: 100%` to the `div` parent container (but this would need to use `View` to work properly), but then I'm unsure if we want to add more DOM elements that the ones we need.

One benefit is that it simplifies the logic for calculating the size as it totally delegates that to the Browser internally.

How do you feel about this approach?

Reply:
To expand on the idea, it would be something like:

```
<View style={{ width: "100%"}}><StyledTextArea ... /></View>
```

Reply:
Ah yes you're right! That works with resizing and it is a simpler solution, thank you for the suggestion! I've made this update!

The parent container is now a View component with `width: 100%`. The View component provides `display: flex` and `align-items: stretch`, which makes the textarea expand to the full width by default. Once the textarea resize control is used, the browser is able to update the width of it since we don't have any `!important` width styling directly on it :) 

> I'm unsure if we want to add more DOM elements that the ones we need.

I think this is okay! I was wrapping the `<textarea>` in a `<div>` anyways so we can still have control over the layout within the TextArea component if we add a character count label for it later on! Then we're less likely to break layouts if we introduce this later on :) https://github.com/Khan/wonder-blocks/pull/2275#discussion_r1680185405